components/script/dom/cssstyledeclaration.rs

```diff
@@ -235,21 +238,88 @@ impl CSSStyleDeclaration {
     }
 
     fn get_computed_style(&self, property: PropertyId) -> DOMString {
-        match self.owner {
+        let node = match self.owner {
             CSSStyleOwner::CSSRule(..) => {
                 panic!("get_computed_style called on CSSStyleDeclaration with a CSSRule owner")
             },
-            CSSStyleOwner::Element(ref el) => {
-                let node = el.upcast::<Node>();
-                if !node.is_connected() {
-                    // TODO: Node should be matched against the style rules of this window.
-                    // Firefox is currently the only browser to implement this.
-                    return DOMString::new();
-                }
-                let addr = node.to_trusted_node_address();
-                window_from_node(node).resolved_style_query(addr, self.pseudo.clone(), property)
+            CSSStyleOwner::Element(ref element) => element.upcast::<Node>(),
+        };
+
+        if !node.is_connected() {
+            // TODO: Node should be matched against the style rules of this window.
+            // Firefox is currently the only browser to implement this.
```

Comment:
This is no longer true, and spec was changed to reflect this. So we can remove this.

Reply:
Thanks for pointing that out. I've opened #26481 for this.